int64 X
int64 Y
int64 Z
---
int64 angle1
int64 angle2
int64 angle3
